admin/admin-service/src/main/java/io/mosip/admin/validator/DocCatCodeValidator.java

```diff
@@ -0,0 +1,50 @@
+package io.mosip.admin.validator;
+
+import java.util.List;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+import io.mosip.admin.bulkdataupload.repositories.DocumentCategoryRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class DocCatCodeValidator implements ConstraintValidator<DocCatCode, String> {
+	
+	private List<String> docCatCode;
+
+	@Autowired
+	private DocumentCategoryRepository documentCategoryRepository;
+	
+	@Override
+	public void initialize(DocCatCode constraintAnnotation) {
+		if(documentCategoryRepository == null){
+			/*         Note: An additional validation was getting triggered by doInvoke() method of 
+			 * 		 RepositoryListItemWriter class with documentCategoryRepository equal to null
+			 *               which is not desired. This if clause is being used to escape that additional 
+			 *               validation step.
+			 */
+			return;
+		}
+		docCatCode = documentCategoryRepository.findAllByIsDeletedFalseOrIsDeletedIsNull();
```

Comment:
This will still not solve the problem. cached only on initialization, what happens to doc categories which is created after initialization ?
Instead, the fix for the ticket should have been a catch clause where the exception is thrown and return a valid error-code.

Reply:
Removed that fix from this pr as it is not priority.